scripts/Muon/GUI/FrequencyDomainAnalysis/FFT/fft_presenter_new.py

```diff
@@ -175,25 +175,26 @@ def calculate_FFT(self):
                 imaginary_workspace_input = run_PaddingAndApodization(imaginary_workspace_padding_parameters)
         else:
             imaginary_workspace_input = None
+            imaginary_workspace_padding_parameters['InputWorkspace'] = ""
 
         fft_parameters = self.get_fft_inputs(real_workspace_input, imaginary_workspace_input, imaginary_workspace_index)
 
         frequency_domain_workspace = convert_to_field(run_FFT(fft_parameters))
-
         self.add_fft_workspace_to_ADS(real_workspace_padding_parameters['InputWorkspace'],
                                       imaginary_workspace_padding_parameters['InputWorkspace'],
                                       frequency_domain_workspace)
 
     def add_fft_workspace_to_ADS(self, input_workspace, imaginary_input_workspace, fft_workspace):
         run = re.search('[0-9]+', input_workspace).group()
-        Im_run = re.search('[0-9]+', imaginary_input_workspace).group()
+        Im_run = ""
+        if imaginary_input_workspace is not "":
+             Im_run = re.search('[0-9]+', imaginary_input_workspace).group()
```

Comment:
```suggestion
            Im_run = re.search('[0-9]+', imaginary_input_workspace).group()
```

Reply:
ah thanks. I will do it later when I fix the unit tests